flutter/lib/src/screenshot/widget_filter.dart

```diff
@@ -80,7 +80,7 @@ class WidgetFilter {
         break;
       case SentryMaskingDecision.continueProcessing:
         // If this element should not be obscured, visit and check its children.
-        element.visitChildElements(_visitList.add);
+        element.debugVisitOnstageChildren(_visitList.add);
```

Comment:
This does not seem to be 100% reliable. So far in my testing, I've seen one occasion when masks were off. Never seen that before. I'll check the implementations of debugVisitOnstageChildren().

In the following screenshot (and the correct one for reference), it seems way off. Note: this has been taken while scrolling rapidly.
<img width="352" alt="image" src="https://github.com/user-attachments/assets/f3b19eb0-d396-45b0-9a18-3d2863f59e99" />
<img width="346" alt="image" src="https://github.com/user-attachments/assets/dd24e493-8654-4c2e-8b82-3ee61bf13bba" />

Reply:
What app are you testing with? I would like to reproduce the same time, but besides that, in my tests if I scroll quickly with visitChildElements the same thing will happen (masks were off).

try comparing the test you just did using visitChildElements and debugVisitOnstageChildren.

Reply:
> What app are you testing with? 
https://github.com/vaind/spotube/tree/sentry

>  in my tests if I scroll quickly with `visitChildElements` the same thing will happen (masks were off).
> try comparing the test you just did using `visitChildElements` and `debugVisitOnstageChildren`.

Interesting, I've never noticed that before. If you have a good repro you could shere for this, I'd like to take a look.